samples/net/azure_iot_hub/boards/nrf9160dk_nrf9160_ns.conf

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# LibC
+CONFIG_PICOLIBC=y
+
+# Heap and stacks
+CONFIG_HEAP_MEM_POOL_SIZE=4096
+CONFIG_MAIN_STACK_SIZE=4096
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
+
+# LED control
+CONFIG_DK_LIBRARY=y
+
+# Disable Duplicate Address Detection (DAD)
+# due to not being properly implemented for offloaded interfaces.
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+
+# Connection Manager and Connectivity layer
+CONFIG_LTE_CONNECTIVITY=y
+CONFIG_NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC=y
+
+# Turn off automatic behavior to give more control to the application
+CONFIG_NRF_MODEM_LIB_NET_IF_AUTO_START=n
+CONFIG_LTE_CONNECTIVITY_AUTO_CONNECT=n
```

Comment:
Could enable auto-connect. This will compress the API calls a bit, making it so that only `conn_mgr_all_if_up()` needs to be called to initialize and bring up the connection. But, I don't really mind separating the two actions in general.